### PR TITLE
Reddit and IMDb link fetcher

### PR DESCRIPTION
**Pull request by claytonmgravatt**

### CODE REVIEW

File: imdbfunction.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        soup = BeautifulSoup(resp.content, from_encoding=encoding,features="lxml")
+        
+        imdbLinkList = []
+        
+        #find links to titles
+        for link in soup.find_all('a', href= re.compile('title')) :
+            #remove irrelevant links
+            if "vote" not in link['href'] and "search" not in link['href'] and "plotsummary" not in link['href'] :
+                #remove duplicates
+                if('https://www.imdb.com' + link['href'] not in imdbLinkList) :
```
Comment (gregv) [inline imdbfunction.py:26]: Optionally, you could just add these to a `set()` and you wouldn't need this if logic

File: redditfunction.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+import requests
+import json
+
+class RedditFunction :
+    
+    #This function returns a list of reddit post links
+    def getPushshiftData(self, size, after, before, queryr) :
+        url = 'https://api.pushshift.io/reddit/search/submission/?&sort=desc&sort_type=score&over_18=false&score=>2000&size=' +str(size) + '&after=' + str(after) + '&before=' +str(before) + '&q=' + str(queryr)
```
Comment (gregv) [inline redditfunction.py]: A little cleaner and more descriptive way of building GET parameters: https://stackoverflow.com/a/44552191

**Review by gregv — APPROVED** [review]
Looks good! Nothing major to call out.

File: imdbfunction.py (hunk 1 above)
Comment (claytonmgravatt) [reply imdbfunction.py:26]: That's a good idea, but the links in this list have a specific order which sets won't guarantee upon creation. I am still open to and looking for better logic for this block of code.

File: redditfunction.py (hunk 2 above)
Comment (claytonmgravatt) [reply redditfunction.py]: Thank you, I may refactor down the line, but for now the URLs will all be consistent so it shouldn't need to check for slashes, but I do like the cleaner arguments layout so I will bookmark this for the future.

**Comment by gregv** [comment]
I'm going to merge in this work so far and close out this Pull Request (PR)

**Pull request merged**